spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/metadata/DefaultManagementMetadataProvider.java

```diff
@@ -24,22 +24,27 @@ public ManagementMetadata get(EurekaInstanceConfigBean instance, int serverPort,
 			return null;
 		}
         String healthCheckUrl = getHealthCheckUrl(instance, serverPort, serverContextPath,
-                managementContextPath, managementPort);
+                managementContextPath, managementPort, false);
         String statusPageUrl = getStatusPageUrl(instance, serverPort, serverContextPath,
                 managementContextPath, managementPort);
 
-        return new ManagementMetadata(healthCheckUrl, statusPageUrl, managementPort == null ? serverPort : managementPort);
+        ManagementMetadata metadata = new ManagementMetadata(healthCheckUrl, statusPageUrl, managementPort == null ? serverPort : managementPort);
+        if(instance.isSecurePortEnabled()) {
+            metadata.setSecureHealthCheckUrl(getHealthCheckUrl(instance, serverPort, serverContextPath,
+                    managementContextPath, managementPort, true));
```

Comment:
Why the new parameter? Isn't it available via `instance.isSecurePortEnabled()`?

Reply:
Yes but if I leave the code as is then both the health check url and the status page url will also be secure.  And I think we wanted to leave the existing functionality as is but just set the correct secure health check url.  So to do this and reuse the code I added the parameter so when we call getHealthCheckUrl and getStatusPageUrl we can explicitly make sure it is not secure even though the secure port is enabled.

Reply:
I see.